sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java

```diff
@@ -144,6 +144,14 @@ public synchronized void start() {
             return;
         }
         logger.info("Starting a new event processor instance.");
+        try {
+            partitionBasedLoadBalancer.checkBlobStatus().block();
+        } catch (RuntimeException ex) {
+            logger.error("Start event processor error", ex);
+            partitionBasedLoadBalancer.removePartitionCache();
```

Comment:
Will partitionBasedLoadBalancer have cache? From code, it is scheduled after this check.

Or put is another way, why not do this in `checkBlobStatus`?

Reply:
partitionBasedLoadBalancer has the partition cache. In check blob status, I'll cache the partition for loadbalancer to use.

I use the partition cache to check did the client start successfully. If client started successfully, then client stop and start again, the client doesn't need to check blob status. If client start failed, I remove the cache.